### PR TITLE
Lpal-398 phase2 removals

### PR DESCRIPTION
**Pull request by williamfalconeruk**
## Purpose
to remove all remaining items that are no longer required:
- VPC
- VPC flow Logs
- Load balancer log bucket
- Networking
- Autoscale roles
- Maintenance page (Legacy)
- Remove unnecessary providers.

we're keeping 
- keys
- secrets
- archive

Secrets and KMS have had some tightening up of security e.g. CMK encryption and auto rotate, and some tfsec warnings ignored. 

We will need to go through the list of secrets to see what can be removed.
We will need to stop the build also on CircleCI once completed through the environments

- Fixes LPAL-398

## Approach

- Deletion of terraform.
- Adjust remaining items.

## Learning
N/A 

## Checklist

* [X] I have performed a self-review of my own code
* [ ] I have updated documentation (Confluence/GitHub wiki/tech debt doc) where relevant
* [ ] I have added tests to prove my work
* [ ] The product team have tested these changes

### CODE REVIEW

File: terraform/account/archive.tf [hunk 1]
```diff
@@ -1,4 +1,5 @@
 #KMS key for Server Side Encryption (SSE) of S3 buckets
+#tfsec:ignore:AWS019
```
Comment (townxelliot) [inline terraform/account/archive.tf:2]: What's this for?

File: terraform/account/credentials.tf [hunk 2]
```diff
@@ -12,55 +12,14 @@ terraform {
...
-}
-variable "old_account_default_role" {
-  default = "opg-refunds-ci"
-}
 
 
 provider "aws" {
-  version = "2.70.0"
-  region  = "eu-west-1"
+  region = "eu-west-1"
```
Comment (townxelliot) [inline terraform/account/credentials.tf:18]: Version no longer required?

File: terraform/account/put_secrets.tf [hunk 3]
```diff
@@ -1,147 +1,184 @@
...
+
 
 
 # public front secrets
 resource "aws_secretsmanager_secret" "opg_refunds_public_front_session_encryption_key" {
-  name = "${local.account_name}/opg_refunds_public_front_session_encryption_key"
-  tags = merge(local.default_tags, local.public_front_component_tag)
+  name       = "${local.account_name}/opg_refunds_public_front_session_encryption_key"
+  tags       = merge(local.default_tags, local.public_front_component_tag)
+  kms_key_id = aws_kms_key.opg_refunds_secrets_key.arn
```
Comment (townxelliot) [inline terraform/account/put_secrets.tf:13]: What's the significance of the kms_key_id entries?

File: terraform/environment/terraform.tfvars.json [hunk 4]
```diff
@@ -53,8 +53,8 @@
       "account_id": "805626386523",
       "is_production": true,
       "aurora_serverless_auto_pause": true,
-      "database_deletion_protection": true,
-      "skip_final_snapshot": false,
+      "database_deletion_protection": false,
```
Comment (townxelliot) [inline terraform/environment/terraform.tfvars.json:56]: It'd be useful to know the reasons why these two have flipped.

File: terraform/account/credentials.tf (hunk 2 above)
Comment (williamfalconeruk) [reply terraform/account/credentials.tf:18]: yep - there's a different way of doing this in the future.

File: terraform/account/archive.tf (hunk 1 above)
Comment (williamfalconeruk) [reply terraform/account/archive.tf:2]: this is a tfsec override,we're trialling this. this override is to ignore the rule for bucket logging. no-one other than ops will have access). see https://github.com/tfsec/tfsec for details of the tool.

File: terraform/account/put_secrets.tf (hunk 3 above)
Comment (williamfalconeruk) [reply terraform/account/put_secrets.tf:13]: these are to tighten up the keys, we now use a Customer Managed Key (not the default AWS one) and is one of the recoomendatins for tfsec. this CMK is auto rotated.

File: terraform/environment/terraform.tfvars.json (hunk 4 above)
Comment (williamfalconeruk) [reply terraform/environment/terraform.tfvars.json:56]: this was preventing deletion of the final db, and we took a manual snapshot as no-one had access. missed that on previous code, so this is to tidy that up (also an explanation - sorry 😁  )

**Review by townxelliot — APPROVED** [review]

**Pull request merged**